src/translator/service.h

```diff
@@ -86,7 +116,9 @@ class Service {
 
   /// Number of workers to launch.
   size_t numWorkers_;        // ORDER DEPENDENCY (pcqueue_)
-  const void *model_memory_; /// Model memory to load model passed as bytes.
+  AlignedMemory modelMemory_; /// Model memory to load model passed as bytes.
```

Comment:
I prefer ORDER DEPENDENCY (also on which other member) being explicit so cautious move around happens. https://isocpp.org/wiki/faq/ctors#order-dependency-in-members. The present ordering is correct. If tomorrow I decollapse Service into a single threaded version (not that this is happening, just one example where swapping unknowingly will create issues), I would like to immediately know that one member has to be initialized before the other when copy-pasting. Both memories should be init before BatchTranslator (this happens already, just indicate ORDER DEPENDENCY explicitly).

We should probably enable `-Werror=reorder` like marian has (but that's for another PR/task).

Reply:
I do not understand here. modelMemory_ and shortlistMemory_ are two independent members. The order of initialisation does not break the constructor. They can be used together or separately, right?

Reply:
Jerin has a peculiar coding style that attempts to annotate whether a subsequent member variable depends on a previous one.  I can see the value in this annotation in ensuring correct destruction order.  In this case he appears to be referring to the workers depending on the continued existence of `modelMemory_`.

Reply:
Both memories are dependent on BatchTranslators (workers are std::threads) just to avoid confusion.